server/src/test/java/org/elasticsearch/indices/state/CloseIndexIT.java

```diff
@@ -0,0 +1,83 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.indices.state;
+
+import io.crate.execution.ddl.tables.TransportCloseTable;
+import io.crate.integrationtests.SQLTransportIntegrationTest;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeUnit;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.index.IndexSettings;
+
+import java.util.List;
+
+import static java.util.Collections.emptySet;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class CloseIndexIT extends SQLTransportIntegrationTest {
+
+    @Override
+    public Settings indexSettings() {
+        return Settings.builder().put(super.indexSettings())
+            .put(IndexSettings.INDEX_TRANSLOG_FLUSH_THRESHOLD_SIZE_SETTING.getKey(),
+                 new ByteSizeValue(randomIntBetween(1, 4096), ByteSizeUnit.KB)).build();
+    }
+
+    /**
+     * Test for https://github.com/elastic/elasticsearch/issues/47276 which checks that the persisted metadata on a data node does not
+     * become inconsistent when using replicated closed indices.
+     */
+    public void testRelocatedClosedIndexIssue() throws Exception {
+        final List<String> dataNodes = internalCluster().startDataOnlyNodes(2);
+        // allocate shard to first data node
+        execute("create table doc.test(x int) clustered into 1 shards with (number_of_replicas=0, \"routing.allocation.include._name\" = ?)", new Object[] {dataNodes.get(0)});
+        var numDocs = randomIntBetween(0, 50);
+        for(var i = 0; i < numDocs; i++) {
+            execute("insert into doc.test values(?)", new Object[]{i});
+        }
+
+        execute("alter table doc.test close");
+        // Closed tables cannot not be altered, therefore use the api
+        client().admin().indices().prepareUpdateSettings("test").setSettings(Settings.builder().put("index.routing.allocation.include._name", dataNodes.get(1))).get();
+        ensureGreen("test");
+        internalCluster().fullRestart();
+        assertIndexIsClosed("test");
```

Comment:
Could we adopt this to do a `sys.shards` query?

Reply:
`assertIndexIsClosed` is asserting against indexMetadata which seems to be not accessible over sql.

Reply:
Given that it is an integration level test I think it would be cool if we could utilize SQL to do the assertion and get the required info. What's currently not exposed from the meta data that would be needed?


Also fine for me to skip this or do it later. Up to you.

Reply:
I will follow up on this.